### PR TITLE
Publish latest documentation

### PR DESCRIPTION
**Pull request by jruzzi**
This PR includes a review of the 2204 changes and fixes files breaking the build. It includes the latest generation script and manually fixes broken cross links to properties.

### CODE REVIEW

File: Messenger/common.json [hunk 1]
```diff
@@ -0,0 +1,190 @@
+{
```
Comment (anand-ky) [inline Messenger/common.json:1]: Can you remove this file. This was updated recently to reflect the latest interface for Messenger.

**Review by anand-ky — CHANGES_REQUESTED** [review]

File: Messenger/common.json (hunk 1 above)
Comment (anand-ky) [reply Messenger/common.json:1]: @npoltorapavlo - Can you check if it's ok to add Common.json file here in Messenger.

File: docs/api/MessengerPlugin.md [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <!-- Generated automatically, DO NOT EDIT! -->
-<a name="head.Messenger_Plugin"></a>
+<a name="Messenger_Plugin"></a>
```
Comment (npoltorapavlo) [inline docs/api/MessengerPlugin.md:2]: "head." and other things are auto-generated. Why are we editing an auto-generated file?

File: Messenger/common.json (hunk 1 above)
Comment (npoltorapavlo) [reply Messenger/common.json:1]: I can't say any of this is "ok" as my understanding is interfaces should be in ThunderInterfaces and not in rdkservices.

Comment (jruzzi) [reply Messenger/common.json:1]: @npoltorapavlo, that is not how rdkservices works at present and I'm not trying to get into a religious war here. The fact is that you added a reference to a file that did not exist and that broken link broke the documentation build. We either have to restore the file the way it was, or we have to add the referenced file (common.json) somewhere in our repository.

File: docs/api/MessengerPlugin.md (hunk 2 above)
Comment (jruzzi) [reply docs/api/MessengerPlugin.md:2]: The generated files are post-processed using a script that fixes the non-standard generated anchors. These anchors lead to broken links in our documentation site.

**Comment by anand-ky** [comment]
Let's go ahead with these changes. Should be fine to have common.json here in RDKServices.

**Comment by jruzzi** [comment]
I will open a separate JIRA to move common.json to a central location so that other plugins can reference the error codes and we do not have to include the error code definitions in each of the JSON definitions.

**Pull request merged**